src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -110,9 +111,21 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.inflightMessages = new InflightMessageList();
 	}
 
+	private Integer getNumPartitions(String topic) {
+		try {
+			return this.kafka.partitionsFor(topic).size(); //returns 1 for new topics
+		} catch(TimeoutException e) {
+			LOGGER.error("Topic '" + topic + "' name does not exist. Exception: " + e.getLocalizedMessage());
+			throw e;
```

Comment:
0.8 doesn't specify its behavior, https://kafka.apache.org/082/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html#partitionsFor(java.lang.String)
0.9 and 0.10 have the same behavior so we have a few more exceptions we should catch here,  https://kafka.apache.org/0100/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#partitionsFor(java.lang.String)
> Throws:
> WakeupException - if wakeup() is called before or while this function is called
> AuthorizationException - if not authorized to the specified topic
> TimeoutException - if the topic metadata could not be fetched before expiration of the configured request timeout
> KafkaException - for any other unrecoverable errors